### PR TITLE
R2RDump - Documentation

### PR DESCRIPTION
**Pull request by acmyu**

### CODE REVIEW

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please

File: src/tools/r2rdump/R2RDump.cs [hunk 1]
```diff
@@ -114,8 +114,6 @@ private ArgumentSyntax ParseCommandLine(string[] args)
                 _sectionContents = true;
             }
 
-            _disasm = false; // TODO: this requires the coredistools nuget package with the most recent changes
```
[the remark this answers is not in the document]
Comment (zamont) [reply src/tools/r2rdump/R2RDump.cs]: I am working on updating the CoreDisTools in parallel. @acmyu can you keep this line and I'll remove it in a follow-up PR?

File: src/tools/r2rdump/Amd64/GcInfo.cs [hunk 2]
```diff
@@ -12,6 +12,9 @@ namespace R2RDump.Amd64
 {
     public class GcInfo : BaseGcInfo
     {
+        /// <summary>
+        /// based on C:\Repos\coreclr\src\inc\gcinfodecoder.h GcInfoHeaderFlags
```
[the remark this answers is not in the document]
Comment (zamont) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: You could paste the entire URL, which will navigate to the gcinfodecoder.h in the master branch: https://github.com/dotnet/coreclr/blob/master/src/inc/gcinfoencoder.h

[the remark this answers is not in the document]
Comment (trylek) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: Sounds good to me. I think there's also some markup for hyperlinks in the XML comments, something like <xref>, looking...

[the remark this answers is not in the document]
Comment (trylek) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: Found it here:

https://stackoverflow.com/questions/6960426/c-sharp-xml-documentation-website-link

///<Summary>
/// This is a math function I found <see href="http://stackoverflow.com">HERE</see>
///</Summary>

[the remark this answers is not in the document]
Comment (zamont) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: Thanks for finding that Tomas!

[the remark this answers is not in the document]
Comment (zamont) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: It actually looks like the <a> tag (a typical HTML link) not <see> works best across tools, according to the StackOverflow discussion.

[the remark this answers is not in the document]
Comment (trylek) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: Hehe, thanks Zach, I noticed the formatting was broken. I actually don't have any idea what is the difference between a "cref" and a "href", perhaps "cref" is more appropriate when referring to code.

[the remark this answers is not in the document]
Comment (trylek) [reply src/tools/r2rdump/Amd64/GcInfo.cs]: OMG, what a mess. I think what matters most is having the link there, having it clickable is a nice-to-have but not an absolute must especially if the silly tagging scheme means we'd need to duplicate the actual URL to have it both as the actual HREF in the A attribute and as the description of the link.

**Comment by trylek** [comment]
This ended up even better than I hoped w.r.t. the links, thanks so much Amy, much appreciated!

**Pull request merged**